### PR TITLE
Refactor analytics

### PR DESCRIPTION
**Pull request by carakas**
I cleaned up the insides of the analytics module a bit.
I don't think I have changed public functions so this shouldn't be a breaking change.

I also made it possible to reset once a .P12 has been uploaded. This makes it easier to reset if a mistake (for instance in the email address) has been made

### CODE REVIEW

File: src/Backend/Core/Engine/Form.php [hunk 1]
```diff
@@ -305,7 +305,7 @@ public function isCorrect($revalidate = true)
     /**
      * Parse the form
      *
-     * @param $tpl The template instance wherein the form will be parsed.
+     * @param $tpl TwigTemplate The template instance wherein the form will be parsed.
```
Comment (WouterSioen) [inline src/Backend/Core/Engine/Form.php]: isn't this ins the wrong order here?

File: src/Backend/Modules/Analytics/Form/DateRangeType.php [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+ */
+final class DateRangeType
+{
+    /** @var Form */
+    private $form;
+
+    /**
+     * @var DateRange $dateRange
+     */
+    private $dateRange;
```
Comment (WouterSioen) [inline src/Backend/Modules/Analytics/Form/DateRangeType.php]: inconsistent `@var` styles

File: src/Backend/Modules/Analytics/Form/SettingsStepType.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+use Backend\Core\Engine\TwigTemplate;
+use Common\ModulesSettings;
+use Google_Service_Analytics;
+
+/**
+ * An interface so we can split the settings form up into multiple forms
+ *
+ * @author Jelmer Prins <jelmer@sumocoders.be>
+ */
+interface SettingsStepType
```
Comment (WouterSioen) [inline src/Backend/Modules/Analytics/Form/SettingsStepType.php]: what's the use case for this interface if you never typehint on it?

File: src/Backend/Modules/Analytics/Form/SettingsStepType.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /**
+     * @param string $name
+     * @param ModulesSettings $settings
+     * @param Google_Service_Analytics $googleServiceAnalytics
+     */
+    public function __construct($name, ModulesSettings $settings, Google_Service_Analytics $googleServiceAnalytics);
```
Comment (WouterSioen) [inline src/Backend/Modules/Analytics/Form/SettingsStepType.php]: I don't think including a constructor in an interface is a best practice.

File: src/Backend/Core/Engine/Form.php (hunk 1 above)
Comment (carakas) [inline src/Backend/Core/Engine/Form.php]: fixed in a3846a0

File: src/Backend/Modules/Analytics/Form/DateRangeType.php (hunk 2 above)
Comment (carakas) [inline src/Backend/Modules/Analytics/Form/DateRangeType.php]: fixed in 5369819

File: src/Backend/Modules/Analytics/Form/SettingsStepType.php (hunk 3 above)
Comment (carakas) [inline src/Backend/Modules/Analytics/Form/SettingsStepType.php]: I do typehint to it https://github.com/forkcms/forkcms/pull/1534/files#diff-93d9fd5c110ab84c96cc9498a4391456R18

File: src/Backend/Modules/Analytics/Form/SettingsStepType.php (hunk 4 above)
Comment (carakas) [inline src/Backend/Modules/Analytics/Form/SettingsStepType.php]: ok, removed in 6dacf34

File: src/Backend/Modules/Analytics/Form/SettingsStepType.php (hunk 3 above)
Comment (WouterSioen) [inline src/Backend/Modules/Analytics/Form/SettingsStepType.php]: a phpdoc is not a typehint ;)

**Comment by WouterSioen** [comment]
merge conflicts alert! ⚠️

**Pull request merged**